### ISSUE TITLE
Starting the bot

### ISSUE THREAD

**Issue by StarryGhost** [Androz2091/discord-music-bot#48]
Hi i keep getting an error 'an application id must be assigned' can someone tell me what im doing wrong

**Comment by elvanz** [comment]
You must assign your BOT's application ID which you can find in https://discord.com/developers/applications

Then, you have to define it in .env (DISCORD_CLIENT_ID = "yourId" 
and call its value to pass in const creator = new SlashCreator(
applicationID = process.env.DISCORD_CLIENT_ID) on index.js

**Comment by StarryGhost** [comment]
what format is the .env file supposed to be... right now i have it as a txt file

**Comment by elvanz** [comment]
.env is already a format you just have to insert all your BOT's credentials into it. I believe there is already one .env.example right here if you fork the project, you just have to remove .example out of .env.example and then do what I said earlier

**Comment by StarryGhost** [comment]
im pretty new to windows and i dont know how to make it just have .env... i copied it over and originally it was an example file...

**Comment by StarryGhost** [comment]
oh wait
i figured it out... now i have another error of the base slash command cant be intantanized

**Comment by elvanz** [comment]
You can do it 2 ways. First, if you are using a text editor like VS Code, you can create a new .env file directly or just rename .env.example to .env. Second, you can navigate directly to the project directory and right click New>Text Document(.txt) and rename that New Text.txt (all of them) to just .env

**Comment by elvanz** [comment]
> 
> 
> oh wait
> i figured it out... now i have another error of the base slash command cant be intantanized

Is your bot already in a server? If it is, you have to add scope commands (oAuth2) bot and application.commands when you invite the BOT to your server

**Comment by StarryGhost** [comment]
my bot is not in a server as of yet should it be?

**Comment by elvanz** [comment]
yes, I think you should invite your bot into a server to debug and test if it works

**Comment by StarryGhost** [comment]
how exactly do i get the bot in?

**Comment by elvanz** [comment]
navigate to discord developer portal here : https://discord.com/developers/applications and then click on your bot application that you have created. From there, you will see OAuth2(it is an url generator for your bot's invitation). Open it and tick 2 boxes in Scopes which are bot and applications.commands and on the Bot Permissions you can add anything you want for you bot. Once you're done, click on copy on the link right below Scopes options and then it will redirect you to discord.

You can find more about it here #2

**Comment by StarryGhost** [comment]
ok i did it and still getting the same error... sorry for all the trouble

**Comment by elvanz** [comment]
it's okay. that would be helpful if you can send me a screenshot of the error

**Comment by StarryGhost** [comment]
https://drive.google.com/file/d/19Zp-zTu6JLEhJIBvKujH7Dtfq7btfzLB/view?usp=sharing

**Comment by StarryGhost** [comment]
both have the same error unfortunately

**Comment by elvanz** [comment]
then it must have something to do with index.js where SlashCommand is not defined properly. if you can, send me a screenshot of your index.js right where you define your slashcommand class

**Comment by StarryGhost** [comment]
https://drive.google.com/file/d/1wGFiAyMPwNz__Mx9cyU0DLFi3eeYbxo7/view?usp=sharing

**Comment by StarryGhost** [comment]
is there any dependancies not listed in the instructions that i have to have?

**Comment by elvanz** [comment]
yeah, it should have no problem tho. i did a fresh reinstall using yarn but failed to generate the same error as yours.

**Comment by elvanz** [comment]
no, just running either yarn install or npm install should install all the dependencies that are registered in package.json.

**Comment by StarryGhost** [comment]
looking at teh firsst issue thing theres the same errors and it was talking about discord.js and Node

**Comment by elvanz** [comment]
oh yeah! i overlooked that issue lol! it has something to do with discordjs version and node.js version. make sure you have the latest discord.js version and node.js version which is 16.9.1 as i recalled

**Comment by StarryGhost** [comment]
how do i check if i have it and the versions

**Comment by elvanz** [comment]
you can check your current node version using node -v

**Comment by StarryGhost** [comment]
my node version is 14.17.6

**Comment by elvanz** [comment]
oh you are running the lts version of node.js. this repo currently only supports the newer version of discordjs(v13) which is using node.js version 16.x

**Comment by elvanz** [comment]
https://nodejs.org/en/ and download the 169.1 latest features and install it again. it will automatically update your current node version

**Comment by StarryGhost** [comment]
aye it started now but its permanently generating docs...

**Comment by elvanz** [comment]
i think the bot is already online! you should ignore the generating docs as it is only a console.log to notify us that the slash-commands has successfully being registered in readme.md

**Comment by StarryGhost** [comment]
but the commands arnt working... *AAAA*

**Comment by elvanz** [comment]
reinvite the bot and you should be fine!

**Comment by StarryGhost** [comment]
the bot is visibly online but non functional...

**Comment by elvanz** [comment]
oh, have you assigned your GUILD_ID in your .env file? if you want your bot to be in multiple servers then delete DISCORD_GUILD_ID in .env and it should sync all the slash commands in each servers. if you just want to run it in a single server then you have to assign your guild(server)id in .env file

**Comment by StarryGhost** [comment]
is there a way to restart the bot to update the file?

**Comment by elvanz** [comment]
press Ctrl + C in the terminal and then type node . again

**Comment by StarryGhost** [comment]
the commands still arnt working

**Comment by elvanz** [comment]
it is because of the made changes. try reinviting the bot again to see if the commands arent still working

**Comment by StarryGhost** [comment]
still no luck

**Comment by StarryGhost** [comment]
does it matter if i start it with yarn or npm

**Comment by elvanz** [comment]
no both yarn start and npm start return node . so it should be fine. try reading #2 and #3 to see if you missed something

**Comment by StarryGhost** [comment]
uh is the guild id the server id?

**Comment by StarryGhost** [comment]
and thanks for all the help

**Comment by elvanz** [comment]
yes it is! to copy the server id you need to enable developer mode in discord setting

**Comment by elvanz** [comment]
> 
> 
> and thanks for all the help

don't mention it. happy to help!

**Comment by StarryGhost** [comment]
ok after trying with a server id... i got a whole lotta green text

**Comment by elvanz** [comment]
does the green text looks like this in #22 ?

**Comment by StarryGhost** [comment]
yes actually

**Comment by elvanz** [comment]
you need to add scope command like this when you invite your bot to your server
![Screenshot 2021-09-20 235126](https://user-images.githubusercontent.com/65339716/134042165-bb6bc0b5-4db8-45b2-964d-f162b7a6ea73.png)

**Comment by StarryGhost** [comment]
those are the scope commands that i added

**Comment by StarryGhost** [comment]
https://drive.google.com/file/d/1QNfmEaPosHXVjoshfSXI3OM6O3GJcwcq/view?usp=sharing

**Comment by elvanz** [comment]
it should already work. if you already added scope commands & also defined GUILD_ID in .env file it should sync slash-commands in your server

**Comment by elvanz** [comment]
maybe try to restart the bot again?

**Comment by StarryGhost** [comment]
... now i get the application id must be defined...

**Comment by elvanz** [comment]
oh gosh xD! how could this happened again.. you made sure you filled in all the credentials in .env file, right?

**Comment by StarryGhost** [comment]
aaaaaa the commands started registering again... the music isnt playing tho... it joined but no sound

**Comment by elvanz** [comment]
have you already installed ffmpeg? if yes, #12 said that you have to install opusscript because of a missing dependency

**Comment by StarryGhost** [comment]
i just have to install ffmpeg with git r?

**Comment by elvanz** [comment]
better to install ffmpeg-static instead by typing yarn add ffmpeg-static or npm install --save ffmpeg-static (choose your own preference whether to use npm or yarn)

**Comment by StarryGhost** [comment]
AAA ITS WORKING TYSM

**Comment by elvanz** [comment]
happy to help!!
